### ISSUE TITLE
Huge fluctuations in number of topics

### ISSUE THREAD

**Issue by tibfab** [MaartenGr/BERTopic#750]
BERTopic v0.12.0

Hi, I'm analyzing customer reviews consisting of relatively short content and found a considerable fluctuation in the number of generated topics if I use BERTopic multiple times in a sequence, like 15, 28, 20, 23, 25...

Is this normal?

I know about the stochastic nature of the approach and don't expect to have the same results, but I try to understand what I might be doing wrong.
Also, after a kernel restart (Jupyter), the results are very stable/similar in contrast to multiple runs in a sequence, as described above. Is there any random seed somewhere in the code?

This is how I fit the model, so it is newly created each time:

    model = BERTopic(language='german')
    model.fit_transform(docs)

Please help me to get a better understanding.

Thanks for your huge effort in this package, making our life easier.

**Comment by MaartenGr** [comment]
This is due to the stochastic nature of UMAP, which is used for the dimensionality reduction part of the pipeline. In order to make sure that you get the same number of topics each time, I would advise using the `random_state` parameter of UMAP as defined [here](https://maartengr.github.io/BERTopic/faq.html#why-are-the-results-not-consistent-between-runs). This also applies when you are using other algorithms instead of UMAP and HDBSCAN. For example, if you are using k-Means instead, you will also need to set the `random_state` there to make sure you get the same output each time.

**Comment by tibfab** [comment]
Great, thanks for pointing me in the right direction.
